### PR TITLE
Fix slot metric query to account for replica slots

### LINKED ISSUE #290: Account for logical replication slots on replicas with patroni
**Describe the issue:**
postgres_exporter gives scrape error warnings on replicas when permanent logical replication slots are in use with patroni.
```
postgres=# SELECT slot_name, active::int, pg_wal_lsn_diff(pg_current_wal_insert_lsn(), restart_lsn) AS retained_bytes FROM pg_catalog.pg_replication_slots;
ERROR:  recovery is in progress
HINT:  WAL control functions cannot be executed during recovery.
```

**Describe the expected behavior:**
Normally there's no slots on a replica, so this would return null there. However, patroni makes "permanent" replication slots that exist on replicas. Either have to remove the retained_bytes data or rewrite the query to work despite it.

**Tell us about your environment:**  
- pgMonitor version:  4.6
- Container or non-container:  
- Container name / image:  
- Operating System for non-container:  R7
- PostgreSQL Version:  13
- Exporter(s) in use (incl. version):  postgres_exporter 0.10.1
- Prometheus version:  
- AlertManager version:  
- Grafana version:

### PR DESCRIPTION
**Pull request by keithf4**
# Description  

Update slot monitoring query to account for logical slots on replicas (patroni feature)

Update changelog and documentation for next release.

Update PR Template & CODEOWNERS

Get pgbouncer fix merged back into development branch. It had gotten merged into main

Fixes #290

Depends on #  

## Type of change  
Please check all options that are relevant  
- [x] Bug fix (change which fixes an issue)  
- [ ] Feature (change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update only  

# How Has This Been Tested?  
**Tested Configuration**:  
- [x] Ansible, Specify version(s):  2.9
- Installation method:  
    - [ ] Binary install from source, version:  
    - [ ] OS package repository, distro, and version:  
    - [ ] Local package server, version:  
    - [ ] Custom-built package, version:  
    - [ ] Other:  
- [x] PostgreSQL, Specify version(s):  12
- [x] docs tested with hugo version(s):  0.80.0

Tested with playbook(s):  

# Checklist:  
- [ ] My changes generate no new lint warnings  
- I have made corresponding changes to:  
    - [ ] the documentation  
    - [ ] the release notes  
    - [ ] the upgrade doc

### CODE REVIEW

File: postgres_exporter/common/pg10/queries_general.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ ccp_replication_lag_size:
 
 
 ccp_replication_slots:
-  query: "SELECT slot_name, active::int, pg_wal_lsn_diff(pg_current_wal_insert_lsn(), restart_lsn) AS retained_bytes FROM pg_catalog.pg_replication_slots"
+  query: "SELECT slot_name, active::int, pg_wal_lsn_diff(CASE WHEN pg_is_in_recovery() THEN pg_last_wal_replay_lsn() ELSE pg_current_wal_insert_lsn() END, restart_lsn) AS retained_bytes"
```
Comment (pgguru) [inline postgres_exporter/common/pg10/queries_general.yml:90]: Did you mean to lose the `FROM` part here?

File: postgres_exporter/common/pg11/queries_general.yml [hunk 2]
```diff
@@ -87,7 +87,7 @@ ccp_replication_lag_size:
 
 
 ccp_replication_slots:
-  query: "SELECT slot_name, active::int, pg_wal_lsn_diff(pg_current_wal_insert_lsn(), restart_lsn) AS retained_bytes FROM pg_catalog.pg_replication_slots"
+  query: "SELECT slot_name, active::int, pg_wal_lsn_diff(CASE WHEN pg_is_in_recovery() THEN pg_last_wal_replay_lsn() ELSE pg_current_wal_insert_lsn() END, restart_lsn) AS retained_bytes"
```
Comment (pgguru) [inline postgres_exporter/common/pg11/queries_general.yml:90]: Same.

**Review by pgguru — CHANGES_REQUESTED** [review]

File: postgres_exporter/common/pg10/queries_general.yml (hunk 1 above)
Comment (keithf4) [reply postgres_exporter/common/pg10/queries_general.yml]: No that certainly wasn't intentional. Thanks!

File: postgres_exporter/common/pg11/queries_general.yml (hunk 2 above)
Comment (keithf4) [reply postgres_exporter/common/pg11/queries_general.yml]: Fixed

**Review by pgguru — DISMISSED** [review]

**Review by pgguru — APPROVED** [review]

**Review by hunleyd — APPROVED** [review]

**Review by jchancojr — APPROVED** [review]

**Pull request merged**